src/smt/witness_form.cpp

```diff
@@ -92,6 +93,17 @@ Node WitnessFormGenerator::convertToWitnessForm(Node t)
           Node skBody = SkolemManager::getSkolemForm(curw[1]);
           Node exists = nm->mkNode(kind::EXISTS, curw[0], skBody);
           ProofGenerator* pg = skm->getProofGenerator(exists);
+          if (pg == nullptr)
+          {
+            // it may be a purification skolem
+            pg = convertExistsInternal(exists);
+            if (pg == nullptr)
+            {
+              Trace("witness-form")
+                  << "WitnessFormGenerator: No proof generator for " << exists
+                  << std::endl;
```

Comment:
Don't we want to use some kind of default proof here?

Reply:
Yes, this is just a warning, we fill in via WITNESS_AXIOM below when none is provided.

Reply:
I mean what is gonna be the justification of the existential? Do I understand correctly that it'll be an open leaf?

Reply:
It's justified by a trusted rule "WITNESS_AXIOM".

Reply:
Ah, I see. I forgot about this feature of `addLazyStep`. Maybe add a comment to clarify this?